Kernel/Arch/x86/common/Spinlock.cpp

```diff
@@ -8,35 +8,31 @@
 
 namespace Kernel {
 
-u32 Spinlock::lock()
+InterruptsState Spinlock::lock()
 {
-    u32 prev_flags = cpu_flags();
+    InterruptsState interrupts_state_before_locking = Processor::interrupts_state();
     Processor::enter_critical();
-    cli();
+    Processor::disable_interrupts();
     while (m_lock.exchange(1, AK::memory_order_acquire) != 0)
         Processor::wait_check();
     track_lock_acquire(m_rank);
-    return prev_flags;
+    return interrupts_state_before_locking;
 }
 
-void Spinlock::unlock(u32 prev_flags)
+void Spinlock::unlock(InterruptsState interrupts_state_to_restore)
```

Comment:
nitpick: I think the "previous" wording made sense, consider something like `previous_state` here instead?

Reply:
Changed the wording to `previous_interrupts_state`.